src/Modules/Components/VCF.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import { compose, setStatic, withState } from 'recompose';
+import { connect } from 'react-redux';
+import { connectModules, registerInputs, registerOutputs } from '../actions';
+import Port from '../../Common/Port';
+import Knob from '../../Common/Knob';
+
+const QUAL_MUL = 30;
+
+class VCF extends Component {
+    constructor(props) {
+        super(props);
+        if (!props.audioContext) throw new Error("audioContext property must be provided");
+
+        this._vcf = props.audioContext.createBiquadFilter();  
+        this._vcf.type = "lowpass";        
+        this.handleFrequencyChange = this.handleFrequencyChange.bind(this);
+        this.handleTypeChange = this.handleTypeChange.bind(this);
+        this.setPitch = this.setPitch.bind(this);
+        this.handleQChange = this.handleQChange.bind(this);
+        this.setQuality = this.setQuality.bind(this);       
+    }
+
+    componentWillMount() {
+        const { id, registerInputs, registerOutputs } = this.props;        
+        registerInputs(id, {
+            Input: {
+                connect: audioNode => audioNode.connect(this._vcf),
+                disconnect: audioNode => audioNode.disconnect(this._vcf)
+            },
+        });
+        registerOutputs(id, {
+           Out: this._vcf
+        });
+    }
+
+    handleFrequencyChange(value) {
+        this.props.setFrequency(Number(value), this.setPitch);
+    }
+
+    setPitch()
+    {
+        const { frequency, audioContext } = this.props;
+        // Clamp the frequency between the minimum value (40 Hz) and half of the
```

Comment:
Do you want to keep all the comments from the example?

Reply:
I'll summarise it into one comment? or no comment at all?

Reply:
whatever. it's just looks too much like copy-paste

Reply:
Actually I like the comments, without them it's hard to understand what's being done and why.